### PR TITLE
Tests if the ContestController works

### PR DESCRIPTION
**Pull request by mvr320**
Currently the card header does not work

Add different contest names

Tried a possible SQL problem, Unicode problem, space problem and
something with really a lot of length

### CODE REVIEW

File: webapp/tests/Controller/Jury/ContestControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+     * @var string baseContest The basecontest we use throughout these tests
+     * @dataProvider provideBasePage
+     */
+    public function testContestsOverview (
+	string $role,
+	int $statuscode,
+	string $baseContest) : void
+    {
+    	static::$roles = [$role];
+        $this->logIn();
```
Comment (nickygerritsen) [inline webapp/tests/Controller/Jury/ContestControllerTest.php:27]: The new tests log in automatically, so i think if you want something like this best is to first log out

File: webapp/tests/Controller/Jury/ContestControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+    /**
+     * Test that jury role can NOT add a new contest
+     */
+    public function testCheckAddContestJury () : void
+    {
+        static::$roles = ['jury'];
+        $this->logIn();
+        $this->verifyPageResponse('GET', static::$baseUrl, 200);
+        $this->assertSelectorNotExists('a:contains('.static::$addButton.')');
```
Comment (nickygerritsen) [inline webapp/tests/Controller/Jury/ContestControllerTest.php:69]: Also check that the /add url gives a 403?

File: webapp/tests/Controller/Jury/ContestControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+     * @var string radioValues The Radio settings of the contest
+     * @var array timeString The time for a certain field as string
+     * @dataProvider provideAddContest
+     */
+     public function testCheckAddContest (
+        string $contest,
+        string $radioValues,
+        array $timeString) : void
+    {
+        $this->logIn();
```
Comment (nickygerritsen) [inline webapp/tests/Controller/Jury/ContestControllerTest.php:84]: This login call should not be needed at all i think

File: webapp/tests/Controller/Jury/ContestControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+    /**
+     * Data provider used to test if we can add contests
+     * - the to be created contest
+     * - radio button boolean values (Bit string)
+     */
+    public function provideAddContest () : \Generator
+    {
+        //* - The different times of the contest
+        //TODO: Try things like: "admin'--"
```
Comment (nickygerritsen) [inline webapp/tests/Controller/Jury/ContestControllerTest.php:157]: I wonder if we should really test these sql injection things. This is one of the reasons we use a framework and not direct queries; the framework takes care of it.

File: webapp/tests/Controller/Jury/ContestControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+    /**
+     * Remove the contest
+     */
+    /* TODO: fix this
+    public function testCheckDeleteContest () : void
+    {
+        $contestShort = 'demo';
+        $contestLong = 'Demo contest';
+        $cid = '2';
```
Comment (nickygerritsen) [inline webapp/tests/Controller/Jury/ContestControllerTest.php:192]: Proposal: use the EntityManagerInterface to get the contest with the given short name?

**Review by nickygerritsen — COMMENTED** [review]
Formatting is....strange but i guess you’ll fix that later.

**Comment by mvr320** [comment]
> 
> 
> Formatting is....strange but i guess you’ll fix that later.

WSL... so yes will fix it later.

File: webapp/tests/Controller/Jury/ContestControllerTest.php (hunk 5 above)
Comment (mvr320) [reply webapp/tests/Controller/Jury/ContestControllerTest.php:192]: Sounds good, do we already do this somewhere? Otherwise it will take me some extra time

Comment (nickygerritsen) [reply webapp/tests/Controller/Jury/ContestControllerTest.php:192]: Maybe not in the tests. I will check when on a bigger device

File: webapp/tests/Controller/Jury/ContestControllerTest.php (hunk 4 above)
Comment (mvr320) [reply webapp/tests/Controller/Jury/ContestControllerTest.php:157]: I agree with you that SQL injection should not happen (and if it does... upstream has to fix it) but they are interesting edge cases which are also teamnames.... but as in code it already fails its a TODO which might actually be removed in the end.

Atleast in my contest I always had people who tried DROP table... and especially in the beginning I was slightly afraid of having to enter these.. (never worked ofcourse)

I'll just remove the TODO for now.

File: webapp/tests/Controller/Jury/ContestControllerTest.php (hunk 1 above)
Comment (mvr320) [reply webapp/tests/Controller/Jury/ContestControllerTest.php:27]: they login as admin I assume?

Comment (nickygerritsen) [reply webapp/tests/Controller/Jury/ContestControllerTest.php:27]: As what you define as the static $roles variable
